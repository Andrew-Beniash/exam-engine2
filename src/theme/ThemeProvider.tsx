import React, {createContext, useContext, useMemo} from 'react';
import {useColorScheme} from 'react-native';
import {
  colors,
  spacing,
  radius,
  elevation,
  typography,
  touchTarget,
  motion,
  contrastPairs,
  type ColorScheme,
  type ExamAccent,
} from './tokens';

interface ThemeContextValue {
  colorScheme: ColorScheme;
  examAccent: ExamAccent;
  colors: {
    primary: string;
    success: string;
    warning: string;
    danger: string;
    info: string;
    bg: string;
    surface01: string;
    surface02: string;
    outline: string;
    textHigh: string;
    textMid: string;
    textLow: string;
    accent: string;
    accentGradient: readonly string[];
  };
  spacing: typeof spacing;
  radius: typeof radius;
  elevation: typeof elevation;
  typography: typeof typography;
  touchTarget: typeof touchTarget;
  motion: typeof motion;
  contrastPairs: typeof contrastPairs.light | typeof contrastPairs.dark;
}

const ThemeContext = createContext<ThemeContextValue | null>(null);

interface ThemeProviderProps {
  children: React.ReactNode;
  examAccent?: ExamAccent;
  forcedColorScheme?: ColorScheme;
}

export const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  examAccent = 'cbap',
  forcedColorScheme,
}) => {
  const systemColorScheme = useColorScheme();
  const colorScheme = forcedColorScheme || systemColorScheme || 'light';

  const theme = useMemo(() => {
    const themeColors = colors[colorScheme];
    const accentConfig = colors.accents[examAccent];

    return {
      colorScheme,
      examAccent,
      colors: {
        // Global semantic colors
        primary: colors.primary,
        success: colors.success,
        warning: colors.warning,
        danger: colors.danger,
        info: colors.info,
        
        // Theme-specific colors
        bg: themeColors.bg,
        surface01: themeColors.surface01,
        surface02: themeColors.surface02,
        outline: themeColors.outline,
        textHigh: themeColors.textHigh,
        textMid: themeColors.textMid,
        textLow: themeColors.textLow,
        
        // Exam-specific accent
        accent: accentConfig.primary,
        accentGradient: accentConfig.gradient,
      },
      spacing,
      radius,
      elevation,
      typography,
      touchTarget,
      motion,
      contrastPairs: contrastPairs[colorScheme],
    };
  }, [colorScheme, examAccent]);

  return (
    <ThemeContext.Provider value={theme}>{children}</ThemeContext.Provider>
  );
};

export const useTheme = (): ThemeContextValue => {
  const theme = useContext(ThemeContext);
  
  if (!theme) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  
  return theme;
};