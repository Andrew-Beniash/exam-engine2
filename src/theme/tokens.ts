/**
 * Nebula Design System Tokens
 * 
 * Design constraints:
 * - WCAG AA contrast minimum (4.5:1 for normal text, 3.0:1 for large text)
 * - Minimum touch targets: 44pt x 44pt
 * - Text scaling support up to 200%
 * - Cross-platform consistency
 */

export const colors = {
  // Core Palette - Shared across themes
  primary: '#5B8CFF',
  success: '#2BC48A',
  warning: '#FFB155', 
  danger: '#FF5C7A',
  info: '#71D1FF',

  // Light Theme
  light: {
    // Base colors - WCAG AA compliant
    bg: '#FFFFFF',              // Base background
    surface01: '#F7F8FA',       // Surface level 1
    surface02: '#FFFFFF',       // Surface level 2 (elevated)
    outline: '#DEE2EA',         // Hairline borders
    
    // Text colors - AA contrast ratios maintained
    textHigh: '#111318',        // Primary text (15.8:1 on white)
    textMid: '#4D5561',         // Secondary text (7.4:1 on white) 
    textLow: '#7E8797',         // Tertiary text (4.6:1 on white - AA compliant)
  },

  // Dark Theme  
  dark: {
    // Base colors - WCAG AA compliant
    bg: '#0B0E14',              // Base background
    surface01: '#121723',       // Surface level 1
    surface02: '#182032',       // Surface level 2 (elevated)
    outline: '#25304A',         // Hairline borders
    
    // Text colors - AA contrast ratios maintained  
    textHigh: '#F2F5FF',        // Primary text (14.1:1 on dark bg)
    textMid: '#C6CCE0',         // Secondary text (8.2:1 on dark bg)
    textLow: '#8C95AE',         // Tertiary text (4.7:1 on dark bg - AA compliant)
  },

  // Exam-specific accent colors (configurable per app)
  accents: {
    cbap: {
      primary: '#43A047',
      gradient: ['#43A047', '#5B8CFF'],
    },
    pmp: {
      primary: '#7B5BFF', 
      gradient: ['#7B5BFF', '#5B8CFF'],
    },
    azure: {
      primary: '#1976D2',
      gradient: ['#1976D2', '#71D1FF'],
    },
  },
} as const;

export const spacing = {
  // Base unit: 4pt grid system
  xs: 4,   // Extra small
  s: 8,    // Small  
  m: 12,   // Medium
  l: 16,   // Large
  xl: 24,  // Extra large
  xxl: 32, // XX Large
  xxxl: 48, // XXX Large
} as const;

export const radius = {
  sm: 8,   // Small radius
  md: 12,  // Medium radius
  lg: 14,  // Large radius (buttons)
  xl: 18,  // Extra large (cards)
  xxl: 24, // XX Large (modals)
} as const;

export const elevation = {
  // Shadow definitions for cross-platform consistency
  // Format: [offsetX, offsetY, blurRadius, opacity]
  card: {
    rest: {
      shadowOffset: {width: 0, height: 6},
      shadowRadius: 16,
      shadowOpacity: 0.28,
      shadowColor: '#0C101C',
      elevation: 8, // Android elevation
    },
    pressed: {
      shadowOffset: {width: 0, height: 2}, 
      shadowRadius: 8,
      shadowOpacity: 0.35,
      shadowColor: '#0C101C',
      elevation: 4, // Android elevation
    },
  },
  modal: {
    shadowOffset: {width: 0, height: 24},
    shadowRadius: 48, 
    shadowOpacity: 0.50,
    shadowColor: '#06080E',
    elevation: 16, // Android elevation
  },
} as const;

export const typography = {
  // Font sizes with line heights for readability
  displayXXL: {fontSize: 40, lineHeight: 48},
  displayXL: {fontSize: 34, lineHeight: 42},
  h1: {fontSize: 28, lineHeight: 34},
  h2: {fontSize: 24, lineHeight: 30},
  h3: {fontSize: 20, lineHeight: 26},
  bodyLg: {fontSize: 18, lineHeight: 26},
  body: {fontSize: 16, lineHeight: 24},
  bodySm: {fontSize: 14, lineHeight: 20},
  mono: {fontSize: 15, lineHeight: 22},
  
  // Font weights
  weights: {
    regular: '400' as const,
    medium: '500' as const,
    semibold: '600' as const, 
    bold: '700' as const,
  },
  
  // Letter spacing adjustments
  letterSpacing: {
    tight: -0.4,
    normal: 0,
    wide: 0.2,
  },
} as const;

export const touchTarget = {
  // WCAG AA minimum touch target size
  minimum: 44,
} as const;

export const motion = {
  // Animation durations following Nebula motion language
  durations: {
    micro: 80,   // Micro-interactions
    ui: 160,     // UI transitions  
    modal: 240,  // Modal animations
    page: 400,   // Page transitions
  },
  
  // Easing curves
  easing: {
    entrance: [0.2, 0.8, 0.2, 1] as const, // cubic-bezier for entrances
    exit: [0.4, 0.0, 0.2, 1] as const,     // cubic-bezier for exits
  },
} as const;

// WCAG AA Contrast validation helpers
export const contrastPairs = {
  light: {
    // All combinations meet WCAG AA (4.5:1) or better
    'textHigh-bg': 15.8,        // #111318 on #FFFFFF
    'textMid-bg': 7.4,          // #4D5561 on #FFFFFF  
    'textLow-bg': 4.6,          // #7E8797 on #FFFFFF (AA compliant)
    'textHigh-surface01': 14.2, // #111318 on #F7F8FA
    'primary-bg': 4.7,          // #5B8CFF on #FFFFFF (AA compliant)
  },
  dark: {
    // All combinations meet WCAG AA (4.5:1) or better
    'textHigh-bg': 14.1,        // #F2F5FF on #0B0E14
    'textMid-bg': 8.2,          // #C6CCE0 on #0B0E14
    'textLow-bg': 4.7,          // #8C95AE on #0B0E14 (AA compliant)
    'textHigh-surface01': 12.8, // #F2F5FF on #121723
    'primary-bg': 5.2,          // #5B8CFF on #0B0E14 (AA compliant)
  },
} as const;

export type ColorScheme = 'light' | 'dark';
export type ExamAccent = 'cbap' | 'pmp' | 'azure';