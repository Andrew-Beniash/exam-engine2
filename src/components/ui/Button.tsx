import React, {useState} from 'react';
import {
  Pressable,
  Text,
  StyleSheet,
  ActivityIndicator,
  ViewStyle,
  TextStyle,
  AccessibilityRole,
} from 'react-native';
import {useTheme} from '../../theme';

export interface ButtonProps {
  /**
   * Button text content
   */
  children: React.ReactNode;
  
  /**
   * Button variant style
   */
  variant?: 'primary' | 'secondary' | 'tertiary' | 'destructive';
  
  /**
   * Button size 
   */
  size?: 'small' | 'medium' | 'large';
  
  /**
   * Disabled state
   */
  disabled?: boolean;
  
  /**
   * Loading state - shows spinner
   */
  loading?: boolean;
  
  /**
   * Press handler
   */
  onPress?: () => void;
  
  /**
   * Accessibility label
   */
  accessibilityLabel?: string;
  
  /**
   * Accessibility hint
   */
  accessibilityHint?: string;
  
  /**
   * Test ID for testing
   */
  testID?: string;
  
  /**
   * Additional styles
   */
  style?: ViewStyle;
}

export const Button: React.FC<ButtonProps> = ({
  children,
  variant = 'primary',
  size = 'medium',
  disabled = false,
  loading = false,
  onPress,
  accessibilityLabel,
  accessibilityHint,
  testID,
  style,
}) => {
  const theme = useTheme();
  const [isPressed, setIsPressed] = useState(false);

  const isDisabled = disabled || loading;

  // Size configurations - ensure minimum 44pt touch target
  const sizeConfig = {
    small: {
      height: Math.max(theme.touchTarget.minimum, 36),
      paddingHorizontal: theme.spacing.m,
      fontSize: theme.typography.bodySm.fontSize,
      minWidth: theme.touchTarget.minimum,
    },
    medium: {
      height: Math.max(theme.touchTarget.minimum, 48),
      paddingHorizontal: theme.spacing.l,
      fontSize: theme.typography.body.fontSize,
      minWidth: theme.touchTarget.minimum,
    },
    large: {
      height: Math.max(theme.touchTarget.minimum, 56),
      paddingHorizontal: theme.spacing.xl,
      fontSize: theme.typography.bodyLg.fontSize,
      minWidth: theme.touchTarget.minimum,
    },
  };

  const currentSize = sizeConfig[size];

  // Variant color configurations
  const getVariantStyles = (): {container: ViewStyle; text: TextStyle} => {
    const baseStyle = {
      borderRadius: theme.radius.lg,
      height: currentSize.height,
      paddingHorizontal: currentSize.paddingHorizontal,
      minWidth: currentSize.minWidth,
      borderWidth: 0,
    };

    switch (variant) {
      case 'primary':
        return {
          container: {
            ...baseStyle,
            backgroundColor: isDisabled 
              ? theme.colors.textLow 
              : isPressed 
                ? theme.colors.accent 
                : theme.colors.primary,
            borderWidth: 0,
          } as ViewStyle,
          text: {
            color: isDisabled 
              ? theme.colors.textMid 
              : theme.colors.bg, // White text on primary bg - meets AA contrast
            fontSize: currentSize.fontSize,
            fontWeight: theme.typography.weights.semibold,
          } as TextStyle,
        };

      case 'secondary':
        return {
          container: {
            ...baseStyle,
            backgroundColor: isPressed ? `${theme.colors.primary}12` : 'transparent',
            borderWidth: 1.5,
            borderColor: isDisabled 
              ? theme.colors.outline 
              : theme.colors.primary,
          } as ViewStyle,
          text: {
            color: isDisabled 
              ? theme.colors.textLow 
              : theme.colors.primary,
            fontSize: currentSize.fontSize,
            fontWeight: theme.typography.weights.semibold,
          } as TextStyle,
        };

      case 'tertiary':
        return {
          container: {
            ...baseStyle,
            backgroundColor: isPressed ? `${theme.colors.primary}08` : 'transparent',
            borderWidth: 0,
          } as ViewStyle,
          text: {
            color: isDisabled 
              ? theme.colors.textLow 
              : theme.colors.primary,
            fontSize: currentSize.fontSize,
            fontWeight: theme.typography.weights.semibold,
          } as TextStyle,
        };

      case 'destructive':
        return {
          container: {
            ...baseStyle,
            backgroundColor: isDisabled 
              ? theme.colors.textLow 
              : isPressed 
                ? `${theme.colors.danger}CC` 
                : theme.colors.danger,
            borderWidth: 0,
          } as ViewStyle,
          text: {
            color: isDisabled 
              ? theme.colors.textMid 
              : theme.colors.bg, // White text on danger bg - meets AA contrast
            fontSize: currentSize.fontSize,
            fontWeight: theme.typography.weights.semibold,
          } as TextStyle,
        };

      default:
        // Fallback to primary variant
        return {
          container: {
            ...baseStyle,
            backgroundColor: isDisabled 
              ? theme.colors.textLow 
              : isPressed 
                ? theme.colors.accent 
                : theme.colors.primary,
            borderWidth: 0,
          } as ViewStyle,
          text: {
            color: isDisabled 
              ? theme.colors.textMid 
              : theme.colors.bg,
            fontSize: currentSize.fontSize,
            fontWeight: theme.typography.weights.semibold,
          } as TextStyle,
        };
    }
  };

  const styles = getVariantStyles();

  const accessibilityRole: AccessibilityRole = 'button';

  return (
    <Pressable
      style={[
        buttonStyles.container,
        styles.container,
        style,
      ]}
      onPress={isDisabled ? undefined : onPress}
      onPressIn={() => !isDisabled && setIsPressed(true)}
      onPressOut={() => setIsPressed(false)}
      disabled={isDisabled}
      accessible={true}
      accessibilityRole={accessibilityRole}
      accessibilityLabel={accessibilityLabel || (typeof children === 'string' ? children : undefined)}
      accessibilityHint={accessibilityHint}
      accessibilityState={{
        disabled: isDisabled,
        busy: loading,
      }}
      testID={testID}>
      
      {loading && (
        <ActivityIndicator
          size="small"
          color={styles.text.color}
          style={buttonStyles.spinner}
        />
      )}
      
      <Text 
        style={[
          buttonStyles.text,
          styles.text,
          buttonStyles.loadingOpacity,
          loading && buttonStyles.loading,
        ]}
        allowFontScaling={true} // Support 200% text scaling
        maxFontSizeMultiplier={2.0}>
        {children}
      </Text>
    </Pressable>
  );
};

const buttonStyles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  loading: {
    opacity: 0.7,
  },
  loadingOpacity: {
    opacity: 1,
  },
  spinner: {
    marginRight: 8,
  },
  text: {
    textAlign: 'center',
  },
});