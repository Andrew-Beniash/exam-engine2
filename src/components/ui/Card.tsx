import React, {useState} from 'react';
import {
  View,
  Pressable,
  StyleSheet,
  ViewStyle,
} from 'react-native';
import {useTheme} from '../../theme';

export interface CardProps {
  /**
   * Card content
   */
  children: React.ReactNode;
  
  /**
   * Whether the card is interactive (pressable)
   */
  interactive?: boolean;
  
  /**
   * Press handler for interactive cards
   */
  onPress?: () => void;
  
  /**
   * Elevation variant - affects shadow depth
   */
  elevation?: 'rest' | 'pressed';
  
  /**
   * Card padding variant
   */
  padding?: 'none' | 'small' | 'medium' | 'large';
  
  /**
   * Accessibility label for interactive cards
   */
  accessibilityLabel?: string;
  
  /**
   * Accessibility hint for interactive cards  
   */
  accessibilityHint?: string;
  
  /**
   * Test ID for testing
   */
  testID?: string;
  
  /**
   * Additional styles
   */
  style?: ViewStyle;
}

export const Card: React.FC<CardProps> = ({
  children,
  interactive = false,
  onPress,
  elevation = 'rest',
  padding = 'medium',
  accessibilityLabel,
  accessibilityHint,
  testID,
  style,
}) => {
  const theme = useTheme();
  const [isPressed, setIsPressed] = useState(false);

  // Determine which elevation to use
  const currentElevation = interactive && isPressed ? 'pressed' : elevation;
  const shadowStyle = theme.elevation.card[currentElevation];

  // Padding configurations
  const paddingConfig = {
    none: 0,
    small: theme.spacing.m,     // 12pt
    medium: theme.spacing.l,    // 16pt (phone) / 20pt would be tablet
    large: theme.spacing.xl,    // 24pt
  };

  const cardStyles = StyleSheet.create({
    container: {
      backgroundColor: theme.colors.surface02,
      borderRadius: theme.radius.xl,
      padding: paddingConfig[padding],
      // Apply shadow/elevation styles
      ...shadowStyle,
    },
  });

  // If not interactive, render as a simple View
  if (!interactive) {
    return (
      <View
        style={[cardStyles.container, style]}
        testID={testID}>
        {children}
      </View>
    );
  }

  // Render as interactive Pressable
  return (
    <Pressable
      style={[cardStyles.container, style]}
      onPress={onPress}
      onPressIn={() => setIsPressed(true)}
      onPressOut={() => setIsPressed(false)}
      accessible={true}
      accessibilityRole="button"
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      testID={testID}>
      {children}
    </Pressable>
  );
};