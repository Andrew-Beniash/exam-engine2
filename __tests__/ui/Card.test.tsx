import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';
import {Text} from 'react-native';
import {Card} from '../../src/components/ui/Card';
import {ThemeProvider} from '../../src/theme/ThemeProvider';

// Test wrapper with theme
const TestWrapper: React.FC<{
  children: React.ReactNode;
  colorScheme?: 'light' | 'dark';
  examAccent?: 'cbap' | 'pmp' | 'azure';
}> = ({children, colorScheme, examAccent}) => (
  <ThemeProvider forcedColorScheme={colorScheme} examAccent={examAccent}>
    {children}
  </ThemeProvider>
);

describe('Card', () => {
  describe('Rendering and Content', () => {
    it('renders children content', () => {
      render(
        <TestWrapper>
          <Card testID="test-card">
            <Text>Card Content</Text>
          </Card>
        </TestWrapper>,
      );

      expect(screen.getByTestId('test-card')).toBeTruthy();
      expect(screen.getByText('Card Content')).toBeTruthy();
    });

    it('renders as non-interactive by default', () => {
      render(
        <TestWrapper>
          <Card testID="non-interactive-card">
            <Text>Non-Interactive Card</Text>
          </Card>
        </TestWrapper>,
      );

      const card = screen.getByTestId('non-interactive-card');
      
      // Should not have button role when not interactive
      expect(card.props.accessibilityRole).not.toBe('button');
      
      // Should render as a View, not Pressable
      expect(card.type).toBe('View');
    });

    it('renders as interactive when specified', () => {
      const mockOnPress = jest.fn();
      
      render(
        <TestWrapper>
          <Card testID="interactive-card" interactive={true} onPress={mockOnPress}>
            <Text>Interactive Card</Text>
          </Card>
        </TestWrapper>,
      );

      const card = screen.getByTestId('interactive-card');
      
      // Should have button role when interactive
      expect(card.props.accessibilityRole).toBe('button');
      
      // Press should work
      fireEvent.press(card);
      expect(mockOnPress).toHaveBeenCalledTimes(1);
    });
  });

  describe('Elevation Prop', () => {
    it('respects rest elevation prop', () => {
      render(
        <TestWrapper>
          <Card testID="rest-elevation-card" elevation="rest">
            <Text>Rest Elevation</Text>
          </Card>
        </TestWrapper>,
      );

      expect(screen.getByTestId('rest-elevation-card')).toBeTruthy();
    });

    it('respects pressed elevation prop', () => {
      render(
        <TestWrapper>
          <Card testID="pressed-elevation-card" elevation="pressed">
            <Text>Pressed Elevation</Text>
          </Card>
        </TestWrapper>,
      );

      expect(screen.getByTestId('pressed-elevation-card')).toBeTruthy();
    });
  });

  describe('Padding Variants', () => {
    it('renders with no padding', () => {
      render(
        <TestWrapper>
          <Card testID="no-padding-card" padding="none">
            <Text>No Padding</Text>
          </Card>
        </TestWrapper>,
      );

      expect(screen.getByTestId('no-padding-card')).toBeTruthy();
    });

    it('renders with small padding', () => {
      render(
        <TestWrapper>
          <Card testID="small-padding-card" padding="small">
            <Text>Small Padding</Text>
          </Card>
        </TestWrapper>,
      );

      expect(screen.getByTestId('small-padding-card')).toBeTruthy();
    });

    it('renders with medium padding (default)', () => {
      render(
        <TestWrapper>
          <Card testID="medium-padding-card">
            <Text>Medium Padding</Text>
          </Card>
        </TestWrapper>,
      );

      expect(screen.getByTestId('medium-padding-card')).toBeTruthy();
    });

    it('renders with large padding', () => {
      render(
        <TestWrapper>
          <Card testID="large-padding-card" padding="large">
            <Text>Large Padding</Text>
          </Card>
        </TestWrapper>,
      );

      expect(screen.getByTestId('large-padding-card')).toBeTruthy();
    });
  });

  describe('Interactive Accessibility', () => {
    it('supports custom accessibility label and hint for interactive cards', () => {
      render(
        <TestWrapper>
          <Card
            testID="custom-a11y-card"
            interactive={true}
            accessibilityLabel="Custom Card Label"
            accessibilityHint="Custom Card Hint">
            <Text>Accessible Card</Text>
          </Card>
        </TestWrapper>,
      );

      const card = screen.getByTestId('custom-a11y-card');
      expect(card.props.accessibilityLabel).toBe('Custom Card Label');
      expect(card.props.accessibilityHint).toBe('Custom Card Hint');
    });
  });

  describe('Theme Snapshots', () => {
    it('matches snapshot in light theme', () => {
      const tree = render(
        <TestWrapper colorScheme="light">
          <Card>
            <Text>Light Theme Card</Text>
          </Card>
        </TestWrapper>,
      );

      expect(tree).toMatchSnapshot('Card-light-theme');
    });

    it('matches snapshot in dark theme', () => {
      const tree = render(
        <TestWrapper colorScheme="dark">
          <Card>
            <Text>Dark Theme Card</Text>
          </Card>
        </TestWrapper>,
      );

      expect(tree).toMatchSnapshot('Card-dark-theme');
    });

    it('matches snapshot with rest elevation', () => {
      const tree = render(
        <TestWrapper>
          <Card elevation="rest">
            <Text>Rest Elevation Card</Text>
          </Card>
        </TestWrapper>,
      );

      expect(tree).toMatchSnapshot('Card-rest-elevation');
    });

    it('matches snapshot with pressed elevation', () => {
      const tree = render(
        <TestWrapper>
          <Card elevation="pressed">
            <Text>Pressed Elevation Card</Text>
          </Card>
        </TestWrapper>,
      );

      expect(tree).toMatchSnapshot('Card-pressed-elevation');
    });

    it('matches snapshot as interactive card', () => {
      const tree = render(
        <TestWrapper>
          <Card interactive={true}>
            <Text>Interactive Card</Text>
          </Card>
        </TestWrapper>,
      );

      expect(tree).toMatchSnapshot('Card-interactive');
    });
  });
});