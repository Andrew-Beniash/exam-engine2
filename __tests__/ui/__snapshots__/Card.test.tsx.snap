// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Theme Snapshots matches snapshot as interactive card: Card-interactive 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 18,
        "elevation": 8,
        "padding": 16,
        "shadowColor": "#0C101C",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.28,
        "shadowRadius": 16,
      },
      undefined,
    ]
  }
>
  <Text>
    Interactive Card
  </Text>
</View>
`;

exports[`Card Theme Snapshots matches snapshot in dark theme: Card-dark-theme 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#182032",
        "borderRadius": 18,
        "elevation": 8,
        "padding": 16,
        "shadowColor": "#0C101C",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.28,
        "shadowRadius": 16,
      },
      undefined,
    ]
  }
>
  <Text>
    Dark Theme Card
  </Text>
</View>
`;

exports[`Card Theme Snapshots matches snapshot in light theme: Card-light-theme 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 18,
        "elevation": 8,
        "padding": 16,
        "shadowColor": "#0C101C",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.28,
        "shadowRadius": 16,
      },
      undefined,
    ]
  }
>
  <Text>
    Light Theme Card
  </Text>
</View>
`;

exports[`Card Theme Snapshots matches snapshot with pressed elevation: Card-pressed-elevation 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 18,
        "elevation": 4,
        "padding": 16,
        "shadowColor": "#0C101C",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.35,
        "shadowRadius": 8,
      },
      undefined,
    ]
  }
>
  <Text>
    Pressed Elevation Card
  </Text>
</View>
`;

exports[`Card Theme Snapshots matches snapshot with rest elevation: Card-rest-elevation 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 18,
        "elevation": 8,
        "padding": 16,
        "shadowColor": "#0C101C",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.28,
        "shadowRadius": 16,
      },
      undefined,
    ]
  }
>
  <Text>
    Rest Elevation Card
  </Text>
</View>
`;
